graphistry/bolt_util.py

```diff
@@ -28,48 +28,94 @@ def to_bolt_driver(driver=None):
 
 #TODO catch additional encodings
 def bolt_graph_to_edges_dataframe(graph):
-    df = pd.DataFrame([
-        util.merge_two_dicts(
-            { key: value for (key, value) in relationship.items() },
+
+    is_neptune=False
+
+    for relationship in graph.relationships:
+        # neptune results not returing element_id, so use id instead
+        is_neptune = not hasattr(relationship, 'element_id')
+        break
+
+    if is_neptune:
+        map_dict_df = pd.DataFrame ([
+            {
+                relationship_id_key:   relationship.id,  # noqa: E241
+                relationship_type_key: relationship.type,  # noqa: E241
+                start_node_id_key:     relationship.start_node.id, # noqa: E241
+                end_node_id_key:       relationship.end_node.id, # noqa: E241
+            }
+            for relationship in graph.relationships])
+    else:
+        map_dict_df = pd.DataFrame ([
             {
                 relationship_id_key:   relationship.element_id,  # noqa: E241
                 relationship_type_key: relationship.type,  # noqa: E241
-                start_node_id_key:     relationship.start_node.element_id,  # noqa: E241
-                end_node_id_key:       relationship.end_node.element_id  # noqa: E241
+                start_node_id_key:     relationship.start_node.element_id, # noqa: E241
+                end_node_id_key:       relationship.end_node.element_id, # noqa: E241
             }
-        )
+            for relationship in graph.relationships])
+
+    df = pd.DataFrame([
+            { key: value for (key, value) in relationship.items() }
         for relationship in graph.relationships
     ])
-    if len(df) == 0:
+
+    joined_df = map_dict_df.join(df)
```

Comment:
not as obvious to me whether to do a merge here, or something inline, like

```python
map_dict_df = pd.DataFrame([
    {
        **{relationship_id_key:   relationship.element_id,  # noqa: E241
           relationship_type_key: relationship.type,  # noqa: E241
           start_node_id_key:     relationship.start_node.element_id, # noqa: E241
           end_node_id_key:       relationship.end_node.element_id},  # noqa: E241
        **{key: value for key, value in relationship.items()}
    }
    for relationship in graph.relationships
])
```

afaict both will be index aligned, so doing column appends after may be just as fast